atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/CheckAndSetResult.java

```diff
@@ -14,12 +14,14 @@
  * limitations under the License.
  */
 
-package com.palantir.atlasdb.keyvalue.impl;
+package com.palantir.atlasdb.keyvalue.api;
 
+import com.palantir.common.annotations.ImmutablesStyles.PackageVisibleImmutablesStyle;
 import java.util.List;
 import org.immutables.value.Value;
 
 @Value.Immutable
+@PackageVisibleImmutablesStyle
 public interface CheckAndSetResult<T> {
```

Comment:
I put it in keyvalue.api, because that's basically what it is. It almost feels like KeyValueService#checkAndSet should be amended to use this.

Reply:
Alright have a look now